configs/defs/elasticsearch.textproto

```diff
@@ -0,0 +1,733 @@
+benchmark_configs: {
+    id: "elasticsearch-xpack"
+    compliance_note: {
+      version: { version: "1.0.0" benchmark_document: "" }
+      title: "Ensure XPack Security Module is enabled"
+      description:
+        "This check verifies whether the XPack security module is enabled in Elasticsearch."
+        "XPack security provides various features such as authentication, role-based access control, and transport layer encryption to "
+        "secure the Elasticsearch cluster. By enabling XPack security, you enhance the overall security posture of your Elasticsearch deployment."
+      rationale:
+        "Enabling the XPack security module is crucial for protecting sensitive data and preventing unauthorized access to your Elasticsearch cluster. "
+        "Without proper security measures in place, your cluster may be vulnerable to unauthorized access, data breaches, or other security incidents. "
+        "By enabling XPack security, you establish a strong foundation for securing your Elasticsearch environment and ensure that only authenticated "
+        "and authorized users have access to the cluster."
+      remediation:
+        "To remediate this issue, follow these steps:"
+        "1. Open the `elasticsearch.yml` configuration file for your Elasticsearch installation."
+        "   - The file is typically located in `/etc/elasticsearch/elasticsearch.yml` on Linux systems or `<Elasticsearch_Home>/config/elasticsearch.yml` on other platforms."
+        "2. Search for the `xpack.security.enabled` setting in the configuration file."
+        "3. If the setting is present and set to `false`, modify the value to `true`."
+        "4. If the setting is not present, add the following line to the configuration file:"
+        "      xpack.security.enabled: true"
+        "5. Save the configuration file."
+        "6. Restart the Elasticsearch service for the changes to take effect."
+        "After performing these steps, XPack security will be enabled in Elasticsearch, providing authentication, access control, and other security features. Remember to configure appropriate user accounts, roles, and permissions within Elasticsearch to ensure secure access to the cluster."
+      cis_benchmark: {
+        profile_level: 1
+        severity: LOW
+      }
+      scan_instructions:
+        # /usr/share/elasticsearch/config/elasticsearch.yml
+        "generic{check_alternatives:{"
+        "  file_checks:{"
+        "    files_to_check:{single_file:{path:\"/usr/share/elasticsearch/config/elasticsearch.yml\"}}"
```

Comment:
I believe it should be possible to add multiple files_to_check entries in which case the check passes if at least one of the files has the specified entry. This way everything can fit into one check_alternative.

Reply:
Since this approach would not work for `NONE_MATCH`, we preferred to use the same rule structure for each instance.

Reply:
Fair enough, we can leave it as is then.